module/os/freebsd/zfs/zfs_vfsops.c

```diff
@@ -1328,7 +1328,7 @@ zfs_mount(vfs_t *vfsp)
 	}
 
 	fetch_osname_options(osname, &checkpointrewind);
-	isctlsnap = (zfsctl_is_node(mvp) && strchr(osname, '@') != NULL);
+	isctlsnap = (mvp != NULL && zfsctl_is_node(mvp) && strchr(osname, '@') != NULL);
```

Comment:
There is a check style complaint about the line you changed. It needs to be wrapped.